### PR TITLE
[CFP-136] [CFP-137] Signpost unused material fees

### PR DESCRIPTION
**Pull request by jrmhaig**
#### What

Add a notice on claims to indicate that unused materials fees may be claimed.

#### Ticket

* [Signpost unused material fees on misc fee page](https://dsdmoj.atlassian.net/browse/CFP-136)
* [Unused material fee flag on summary page](https://dsdmoj.atlassian.net/browse/CFP-137)

#### Why

Providers are being underpaid for work they are entitled to claim for.

#### How

A notice is added in the following places of a claim that is eligible for an unused materials fee:

* At the top of the claim summary
* In the claim summary, at the miscellaneous page section
* On the miscellaneous fees page

### CODE REVIEW

File: app/helpers/claims_helper.rb [hunk 1]
```diff
@@ -40,4 +40,21 @@ def messaging_permitted?(message)
...
+    }.tap do |headings|
+      if display_unused_materials_notice_for(claim)
+        headings[:page_notice] = t('unused_materials_fee.notice', scope: scope)
+      end
+      headings[:fees_calculator_html] = fees_calculator_html unless fees_calculator_html.nil?
+    end
+  end
+
+  def display_unused_materials_notice_for(claim)
+    claim.eligible_misc_fee_types.map(&:unique_code).include?('MIUMU') &&
```
Comment (jsugarman) [inline app/helpers/claims_helper.rb:69]: 👍

File: app/helpers/claims_helper.rb [hunk 2]
```diff
@@ -40,4 +40,21 @@ def messaging_permitted?(message)
...
+      page_hint: t('page_hint', scope: scope)
+    }.tap do |headings|
+      if display_unused_materials_notice_for(claim)
+        headings[:page_notice] = t('unused_materials_fee.notice', scope: scope)
+      end
+      headings[:fees_calculator_html] = fees_calculator_html unless fees_calculator_html.nil?
+    end
+  end
+
+  def display_unused_materials_notice_for(claim)
```
Comment (jsugarman) [inline app/helpers/claims_helper.rb]: should this not be `display_unused_materials_notice?`

File: app/services/claims/fetch_eligible_misc_fee_types.rb [hunk 3]
```diff
@@ -5,8 +5,8 @@ def initialize(claim)
     end
 
     def call
-      return nil unless claim
-      return nil if claim&.interim?
+      return unless claim
+      return [] if claim&.interim?
```
Comment (jsugarman) [inline app/services/claims/fetch_eligible_misc_fee_types.rb:9]: into scope creep territory with the commit that promoted these methods and amends this. Are you confident changing this behaviour is not going to break other areas of the app. 
Not sure feature tests and other integration specs cover it so have will need manual testing to check that it does not change behaviour in both the form steps and the various summary pages for all claim types (interims and transfers in particular) that may rely on the existing eligibility logic.

File: app/helpers/claims_helper.rb (hunk 2 above)
Comment (jsugarman) [reply app/helpers/claims_helper.rb]: have we missed copying the tests for it?

File: app/services/claims/fetch_eligible_misc_fee_types.rb (hunk 3 above)
Comment (jrmhaig) [reply app/services/claims/fetch_eligible_misc_fee_types.rb:9]: The `display_unused_materials_notice?` helper depends on the `eligible_misc_fee_types` method existing for any claim type and it must return an array. The correct response for an interim claim is an empty array as it is not eligible for any miscellaneous fee types.

**Review by jsugarman — APPROVED** [review]
👍 good to go

**Pull request merged**